### PR TITLE
NEXT-14408 - Allow CMS blocks to be added by apps

### PR DESCRIPTION
**Pull request by fschmtt**
Add documentation about how to add custom blocks to Shopware's CMS using an app.

### CODE REVIEW

File: guides/plugins/apps/content/cms/add-custom-cms-blocks.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+As already mentioned above and similarly to an app's `manifest.xml` CMS blocks also require some definition which is done in a `cms.xml`.
+In our example we will define two custom CMS blocks:
+
+{% code title="<app root>/Resources/cms.xml" %}
+```xml
+<?xml version="1.0" encoding="utf-8" ?>
+<cms xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/platform/master/src/Core/Framework/App/Cms/Schema/cms-1.0.xsd">
+    <blocks>
+        <block>
+            <name>first-block-name</name> <!-- A unique technical name for your block -->
```
Comment (keulinho) [inline guides/plugins/apps/content/cms/add-custom-cms-blocks.md:54]: can you add a description of all fields under references section? There should already be a reference for the manifest xml that describe all possible configs there, i think a similiar format for cms.xml makes sense

File: guides/plugins/apps/content/cms/add-custom-cms-blocks.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+Each slot has a **unique** `name` and a `type` that refers to which element it shows.
+All available elements can be found in [src/Administration/Resources/app/administration/src/module/sw-cms/elements](https://github.com/shopware/platform/tree/trunk/src/Administration/Resources/app/administration/src/module/sw-cms/elements).
+At a later point you will also be able to define custom elements but for now you can use the elements shipped by Shopware.
+
+The `config` of a slot is very dynamic as it highly depends on which `type` you have chosen.
+
+## Registering blocks
+
+Blocks will be automatically registered during runtime - so all you have to take care of is to properly them.
```
Comment (keulinho) [inline guides/plugins/apps/content/cms/add-custom-cms-blocks.md:153]: here are some words missing :laughing:

File: guides/plugins/apps/content/cms/add-custom-cms-blocks.md (hunk 1 above)
Comment (fschmtt) [reply guides/plugins/apps/content/cms/add-custom-cms-blocks.md:54]: Sure, it's here: resources/references/app-reference/cms-reference.md

File: guides/plugins/apps/content/cms/add-custom-cms-blocks.md (hunk 2 above)
Comment (fschmtt) [reply guides/plugins/apps/content/cms/add-custom-cms-blocks.md]: Whoops, fixed!

File: guides/plugins/apps/content/cms/add-custom-cms-blocks.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+```
+
+Each CMS block defined within your `cms.xml` must have a directory matching the block's name in `Resources/cms/blocks/`.
+In those directories you shape your blocks by supplying a `preview.html` containing the template used for displaying a preview.
+Styling the preview is possible from the `styles.css`.
+
+{% hint type=info %}
+Due to technical limitations it's not possible to use templating engines \(like Twig\) for the preview template or preprocessors \(like Sass\) for styling.
+{% endhint %}
+
```
Comment (SebastianFranze) [inline guides/plugins/apps/content/cms/add-custom-cms-blocks.md:42]: We should tell the app manufacturers that they still have to add the storefront representation of their blocks to the `Resources/views/storefront/cms/block/` folder.

Comment (fschmtt) [reply guides/plugins/apps/content/cms/add-custom-cms-blocks.md:48]: Added section `## Storefront representation` with a short description and linking to the plugin example.

**Review by SebastianFranze — APPROVED** [review]

**Comment by PaddyS** [comment]
Can you please add the entries to the SUMMARY.md file? A new entry for each new directory and file, please :-)

**Review by PaddyS — CHANGES_REQUESTED** [review]
Can you please add the entries to the [SUMMARY.md](https://github.com/shopware/docs/blob/master/SUMMARY.md) file? A new entry for each new directory and file, please :-)

**Pull request merged**